stingray/powerspectrum.py

```diff
@@ -284,6 +284,82 @@ def classical_significances(self, threshold=1, trial_correction=False):
 
         return pvals
 
+    def modulation_upper_limit(self, fmin=None, fmax=None, c=0.95):
+        """Upper limit on a sinusoidal modulation.
+
+        To understand the meaning of this amplitude: if the modulation is described by:
+
+        ..math:: p = \overline{p} (1 + a * \sin(x))
+
+        this function returns a.
+
+        If it is a sum of sinusoidal harmonics instead
+        ..math:: p = \overline{p} (1 + \sum_l a_l * \sin(lx))
+        a is equivalent to :math:`\sqrt(\sum_l a_l^2)`.
+
+        See `stingray.stats.power_upper_limit`, `stingray.stats.amplitude_upper_limit`
+
+        Parameters
+        ----------
+        fmin: float
+            The minimum frequency to search (defaults to the first nonzero bin)
+
+        fmax: float
+            The minimum frequency to search (defaults to the Nyquist frequency)
+
+        Other Parameters
+        ----------------
+        c: float
+            The confidence value for the upper limit (e.g. 0.95 = 95%)
+
+        Results
+        -------
+        a: float
+            The modulation amplitude that could produce P>pmeas with 1 - c probability
+
+        Examples
+        --------
+        >>> pds = Powerspectrum()
+        >>> pds.norm = "leahy"
+        >>> pds.freq = np.arange(0, 5)
+        >>> pds.power = np.array([100000, 5, 2, 40, 1])
+        >>> pds.m = 1
+        >>> pds.nphots = 30000
+        >>> aup = pds.modulation_upper_limit(2, 5, 0.99)
+        >>> aup_corr = amplitude_upper_limit(40, 30000, 1, 0.99, nyq_ratio=3/4, fft_corr=True)
```

Comment:
So I find this example very confusing, because I don't actually know what the difference is between `modulation_upper_limit` and `amplitude_upper_limit`. I think the idea with examples is to give the user a quick overview of the correct usage. I think it'd be helpful if it had slightly more explanations, and maybe left out the comparison for easier comprehension?